src/renderer/port-forward/port-forward.store.ts

```diff
@@ -0,0 +1,218 @@
+  port: string;
+  forwardPort: string;
+}
+
+export class PortForwardStore extends ItemStore<PortForwardItem> {
+  private storage = createStorage<ForwardedPort[] | undefined>("port_forwards", undefined);
+
+  constructor() {
+    super();
+    makeObservable(this);
+    autoBind(this);
+
+    this.init();
+  }
+
+  private async init() {
+    await this.storage.whenReady;
+
+    const savedPortForwards = this.storage.get(); // undefined on first load
+
+    if (Array.isArray(savedPortForwards)) {
+      logger.info("[PORT_FORWARD] starting saved port-forwards");
+      await Promise.all(savedPortForwards.map(pf => {
+        const port = new PortForwardItem;
+
+        port.clusterId = pf.clusterId;
+        port.kind = pf.kind;
+        port.namespace = pf.namespace;
+        port.name = pf.name;
+        port.port = pf.port;
+        port.forwardPort = pf.forwardPort;
+
+        return addPortForward(port);
+      }));
+
+      this.portForwards = [];
+    }
+  }
+
+  @observable selectedItemId?: string;
+  @observable portForwards: PortForwardItem[];
+
+  @computed get selectedItem() {
+    return this.portForwards.find((e: ItemObject) => e.getId() === this.selectedItemId);
+  }
+
+  watch() {
+    const disposers: IReactionDisposer[] = [
+      reaction(() => this.portForwards, () => this.loadAll()),
+    ];
+
+    return () => disposers.forEach((dispose) => dispose());
+  }
+
+  loadAll() {
+    return this.loadItems(async () => {
+      let portForwards = await getPortForwards();
+
+      // filter out any not for this cluster
+      portForwards = portForwards?.filter(pf => pf.clusterId == getHostedClusterId());
+      this.storage.set(portForwards);
+
+      this.reset();
+      portForwards?.forEach(pf => {
+        const port = new PortForwardItem;
+
+        port.clusterId = pf.clusterId;
+        port.kind = pf.kind;
+        port.namespace = pf.namespace;
+        port.name = pf.name;
+        port.port = pf.port;
+        port.forwardPort = pf.forwardPort;
+        this.portForwards.push(port);
+      });
+
+      return this.portForwards;
+    });
+  }
+
+  reset() {
+    this.portForwards = [];
+  }
+
+  async removeSelectedItems() {
+    return Promise.all(this.selectedItems.map(removePortForward));
+  }
+}
+
+interface PortForwardResult {
+  port: number;
+}
+
+interface PortForwardsResult {
+  portForwards: ForwardedPort[];
+}
+
+export async function addPortForward(portForward: ForwardedPort): Promise<number> {
+  let response: PortForwardResult;
+
+  try {
+    response = await apiBase.post<PortForwardResult>(`/pods/${portForward.namespace}/${portForward.kind}/${portForward.name}/port-forward/${portForward.port}/${portForward.forwardPort}`, {});
+    
+    if (response?.port && response.port != +portForward.forwardPort) {
+      logger.warn(`specified ${portForward.forwardPort} got ${response.port}`);
+    }
+  } catch (error) {
+    logger.warn(error); // don't care, caller must check 
+  }
+  portForwardStore.reset();
```

Comment:
Why are we resetting the store after a successful (or for that matter unsuccessful) add?

Reply:
this is how the list knows to refresh. Maybe there is a more efficient way. But the port-forwards are managed in main, and more than one cluster could be trying to add/remove/etc them at the same time. Thought it is safer to just "loadAll"

Reply:
Do we have that view in this PR? I did not see it. Nevertheless I think that think should really be an actual watch with either polling or server push but that is too big for this PR. As long as we don't expose any of this to the extension API I am fine with leaving this as it is for the time being.